File: Javascript/String/Merge-Strings-Alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
const mergeAlternately = (word1, word2) => {
  let [p1, p2] = [0, 0];

  let result = "";

  while (p1 < word1.length && p2 < word2.length) {
    result += word1[p1] + word2[p2];
    p1++;
    p2++;
  }

  if (p1 < word1.length) result += word1.slice(p1);

  if (p2 < word2.length) result += word2.slice(p2);

  return result;
};

File: Rust/Merge-Strings-Alternately.rs
pub struct Solution {}

impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut result = String::new();
        let (mut word1_count, mut word2_count) = (0, 0);
        word1
            .chars()
            .into_iter()
            .zip(word2.chars().into_iter())
            .for_each(|(first, second)| {
                word1_count += 1;
                word2_count += 1;
                result.push(first);
                result.push(second);
            });

        if word1_count < word1.len() {
            result.push_str(&word1[word1_count..]);
        }

        if word2_count < word2.len() {
            result.push_str(&word2[word2_count..]);
        }
        result
    }
}
fn main() {}
